### DISCUSSION TITLE
IMPORTANT NOTICE FOR PUBLIC INSTALLATIONS

### DISCUSSION THREAD

**Discussion by ehpersonal38** [mcguirepr89/BirdNET-Pi#902] [General]
Hi, I've recently implemented a critical patch that all public-facing BirdNET-Pi installations should update to. You can do so in 

**Tools -> Settings -> System Controls -> Update.**

There's a small chance that a vulnerability was exploited which allowed for remote code execution on your Pi. I urge anyone with a public facing installation to update their installations and then check for unusual running processes (`ps -aux`) and suspicious network activity (`netstat -an`). If in doubt, a fresh reimage is always recommended. Better safe than sorry. 

Again, this only applies to installations that are public facing, i.e. have been port forwarded and are accessible on the internet. 

I will be taking measures to ensure that this does not happen again in the future. 

Thank you.

**Comment by becausebirds** [comment]
Hi, I just updated my public installation. What type of unusual processes should I be looking for? I ran both commands and they both returned a bunch of stuff.

**Reply by ehpersonal38** [reply]
I'm not exactly sure. Theoretically this vulnerability would allow someone to wget a bash script or binary onto your Pi and run it as sudo. 

Check to make sure nothing suspicious is in this startup file:
`cat /etc/rc.local`

Check to make sure there are no suspicious services:
`ls -lh /etc/systemd/system`

Make sure there are no suspicious cron entries:
`cat /etc/crontab`
and
`crontab -l`
and make sure there are only 2 files here, one with your username and one with root:
`sudo ls -lh /var/spool/cron/crontabs/`
`

**Reply by oliverk71** [reply]
My birdnetpi installation seems broken. I don't know the reason. 
What if there are no files in `/var/spool/cron/crontabs` ?
`systemctl | grep failed` shows the following: 
```
# caddy.service               loaded failed       failed       Caddy
# hciuart.service               loaded failed       failed       Configure Bluetooth Modems connected by UART
# icecast2.service               loaded failed       failed       LSB: Icecast2 streaming media server
# php7.4-fpm.service              loaded failed       failed       Load/Save RF Kill Switch Status
# user@1000.service               loaded failed       failed       User Manager for UID 1000
# wpa_supplicant.service               loaded failed       failed       WPA supplicant
```
What can I do now?

**Reply by ehpersonal38** [reply]
What's in /var/log/syslog? Any obvious errors? If not then I recommend a reimage. Make sure you backup the following files and directories
```
~/BirdNET-Pi/birdnet.conf
~/BirdNET-Pi/apprise.txt
~/BirdNET-Pi/scripts/birds.db
~/BirdSongs/Extracted/By_Date
~/BirdSongs/Extracted/Charts
```

**Reply by oliverk71** [reply]
a whole lot of kernel stuff

**Reply by oliverk71** [reply]
Currently doing new install.

**Reply by oliverk71** [reply]
After new install it is still very slow and hard to reach over wifi network. Although I copied the database and config and other files, it now has English bird names and no data except from yesterday and today. Maybe it doesn't make much sense on an Rpi 3B, since I had so many problems and repeatedly lost data. Hope, I will have time tomorrow to see what is going on. 
But I guess it has not been hacked and this is maybe good news.

**Reply by lloydbayley** [reply]
Doesn't sound like a hack. There's something weird going on. I had a test install running quite happily for some months on a 3B+.
English bird names you say?! Is the latitude and longitude correct in the settings and what is the database language set to?

**Reply by oliverk71** [reply]
Well, I did not yet had much time to investigate on this. So maybe I made some mistake. However, this branch of the discussion should be moved into a new discussion or can be deleted because it has nothing to do with the topic.

I finally managed to change language settings by editing the birdnet.conf and then change it in the settings page. However, it did not work to change it to German via settings page. Weird, because I was able to change the language, but not to German. Although it was "de" in the conf and the settings page. 

Website SSL did not work, no https available. After I changed the domain name from grevenbroich-elsen.birdnetpi.com to birdnetgv.ddnss.de (which is my old dyndns domain) everything was okay again.  

So, I got it working again. Anyway, I would be very happy, if I someday can get a 4B for a reasonable price.

**Reply by lloydbayley** [reply]
Ahh....Well that's good. Yes, the birdnetpi.com domain is down and all the subdomains are not working. There are a few threads about that. Very glad to hear you got it working now. 
The bit about the language not changing is weird...I wonder if it could be tied into the non-existant domain...
Still, keep an eye on it and see how it goes. If something occurs, just start another discussion and we'll jump in there. :)

**Comment by jds3d** [comment]
Is the birdnetpi domain working?  I haven't been able to access it (or my installation publically) for a while--https://viennava.birdnetpi.com/.

**Comment by lloydbayley** [comment]
There's some transfer going on. See #882.
